### ISSUE TITLE
Add option to avoid HTTP request in Client constructor

### ISSUE THREAD

**Issue by Holt59** [ezhov-evgeny/webdav-client-python-3#29]
Since [this commit](https://github.com/ezhov-evgeny/webdav-client-python-3/commit/5f84bb1343a64316a443164ecde6830ef3c22e3b#diff-174800eb1e3649a0fcf9d54c6ec56571), the `Client` constructor performs a request to the WebDAV server on construction.

This is pretty annoying as it now breaks a lot of unit tests on my side where I used dumb URLs for wrapper object, and I am sure it will impact other users as well.

Would it be possible to add a constructor argument to bypass this?

**Comment by evgeny-ezhov** [comment]
Can you provide an example of dumbing URLs?
And why it is impossible to add dumb for method OPTIONS with return 200 and HEADER “Allow: HEAD”?

**Comment by Holt59** [comment]
Any URL that does not exists, e.g.:

```
client = Client({
    "webdav_hostname": "https://webdav",
    "webdav_user": "user",
    "webdav_login": "login"
})
```

I have classes that have a `_client` member which is a `Client` object constructed during initialization, e.g.:

```
class A:
    def __init__(self, url):
        self._client = Client({"webdav_hostname": url})
```

Before this change, it was possible to perform unit-testing on class `A` using dumb URL because the `Client` constructor did not perform any operation:

```
def test_A_foo():
    a = A(url="https://webdav")
    assert a.foo() == 3
```

Now it is impossible because the `A(...)` initialization will fail, and setting-up a WebDAV client for unit-testing is kind of overkill, especially if I do not use it.

**Comment by ezhov-evgeny** [comment]
In your case, I can suggest trying to use [`patch` from unittest.mock](https://docs.python.org/3/library/unittest.mock.html#patch) like:
```python
@mock.patch("webdav3.client.Client")
def test_A_foo(mock_client):
    a = A(url="https://webdav")
    assert a.foo() == 3
```
It mocks the webdav3.client.Client in this method and put mock object as argument.
I think it solves your problem.

**Comment by ezhov-evgeny** [comment]
@Holt59 Is your problem resolved?

**Comment by Holt59** [comment]
@ezhov-evgeny I'm sorry, I'm not using `unittest` but `pytest` and I did not have time to check if `pytest` offers something similar.

Currently, I've switched-back to the `0.13` version of the package.

But unit-testing aside, I think it would be a good thing to not perform a HTTP request in the `Client` constructor, or at least document it a bit more and allow to defer it.

**Comment by ezhov-evgeny** [comment]
For pytest I can suggest you https://github.com/pytest-dev/pytest-mock/

Actually now I think this request should be in the constructor but I'll think about it

**Comment by ezhov-evgeny** [comment]
@Holt59 I have changed a way to support servers that do not support HEAD method, now there is no requests in the constructor.

**Comment by ezhov-evgeny** [comment]
@Holt59 Just released 3.14 try this version
